conjure-java-jaxrs-client/src/main/java/com/palantir/conjure/java/client/jaxrs/AbstractFeignJaxRsClientBuilder.java

```diff
@@ -119,13 +119,15 @@ private Contract createContract() {
     }
 
     private static Decoder createDecoder(ObjectMapper objectMapper, ObjectMapper cborObjectMapper) {
-        return new ConjureNeverReturnNullDecoder(
-                new ConjureJava8OptionalAwareDecoder(
-                        new ConjureGuavaOptionalAwareDecoder(
-                                new ConjureEmptyContainerDecoder(
-                                        objectMapper,
-                                        new ConjureInputStreamDelegateDecoder(
-                                                new ConjureTextDelegateDecoder(
+        return new ConjureJava8OptionalAwareDecoder(
+                new ConjureGuavaOptionalAwareDecoder(
+                        // The ConjureTextDelegateDecoder is allowed to return null
+                        // This is to support older endpoints that are not conjure-generated
+                        new ConjureTextDelegateDecoder(
+                                new ConjureNeverReturnNullDecoder(
```

Comment:
I swapped the order round this way because:

* optional decoders need to first check if a 204 is compatible with an optional type
* conjure text delegate decoder _only_ handles 204 + text/plain + **return type you can assign string to** (String, CharSequence, Object) -- there are tests for these. the return type check has been added in this PR
* never null decoder handles all other unexpected nulls coming from jackson